lib/combine_pdf/page_methods.rb

```diff
@@ -63,15 +63,16 @@ def inject_page(obj, top = true)
       # injecting each of the values in the injected Page
       res = resources
       obj.resources.each do |key, new_val|
-        unless PDF::PRIVATE_HASH_KEYS.include? key # keep CombinePDF structual data intact.
-          if res[key].nil?
-            res[key] = new_val
-          elsif res[key].is_a?(Hash) && new_val.is_a?(Hash)
-            new_val = new_val[:referenced_object] || new_val
-            new_val.update (res[key][:referenced_object] || res[key]) # make sure the old values are respected
-            (res[key][:referenced_object] || res[key]).update new_val # transfer old and new values to the injected page
-          end #Do nothing if array - ot is the PROC array, which is an issue
-        end
+        # keep CombinePDF structural data intact.
+        next if PDF::PRIVATE_HASH_KEYS.include?(key)
```

Comment:
I love Rubocop, but I haven’t really used its autocorrect features. They
make me very nervous!

On Mon, Jul 18, 2016 at 11:59 AM, Bo <notifications@github.com> wrote:

> In lib/combine_pdf/page_methods.rb
> <https://github.com/boazsegev/combine_pdf/pull/76#discussion_r71174976>:
>
> > @@ -63,15 +63,16 @@ def inject_page(obj, top = true)
> >        # injecting each of the values in the injected Page
> >        res = resources
> >        obj.resources.each do |key, new_val|
> > -        unless PDF::PRIVATE_HASH_KEYS.include? key # keep CombinePDF structual data intact.
> > -          if res[key].nil?
> > -            res[key] = new_val
> > -          elsif res[key].is_a?(Hash) && new_val.is_a?(Hash)
> > -            new_val = new_val[:referenced_object] || new_val
> > -            new_val.update (res[key][:referenced_object] || res[key]) # make sure the old values are respected
> > -            (res[key][:referenced_object] || res[key]).update new_val # transfer old and new values to the injected page
> > -          end #Do nothing if array - ot is the PROC array, which is an issue
> > -        end
> > +        # keep CombinePDF structural data intact.
> > +        next if PDF::PRIVATE_HASH_KEYS.include?(key)
>
> You're saying Rubocop messed up my logic... again... 😖 I'll go look at
> the pre-beautification logic. Thanks.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/boazsegev/combine_pdf/pull/76/files/9e1ecc32e091dc10242a543d5c8570a3fb470e15..688740f2d66b33fd784e401d0ffde5a8b8112b14#r71174976>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAMIN7rrfStmdnvlAovzsIdPTQ3Rmqvpks5qW6LMgaJpZM4JO0PP>
> .
>